rpcq/_base.py

```diff
@@ -109,7 +110,8 @@ def __eq__(self, other):
     def __hash__(self):
         return hash((self.__class__, astuple(self)))
 
-    _types = {}
+    _types = None
+    _types_lock = threading.Lock()
 
     @staticmethod
     def types():
```

Comment:
Wouldn't [functools.lru_cache](https://docs.python.org/3/library/functools.html#functools.lru_cache) be faster & easier here? Pretty sure it'll work with a static method. Also pretty sure it's thread-safe.

Reply:
I think it's thread-safe: https://github.com/python/cpython/blob/3.6/Lib/functools.py#L470

Reply:
Also, it was briefly not thread-safe and it was considered a bug which was fixed: https://bugs.python.org/issue28969

Reply:
OK 👍 Will adjust